packages/modular-scripts/src/fetch.ts

```diff
@@ -0,0 +1,25 @@
+import fetch, { RequestInit } from 'node-fetch';
+import HttpsProxyAgent = require("https-proxy-agent")
+
+const getAgent = () => {
+  if (process.env.HTTP_PROXY || process.env.HTTPS_PROXY) {
+    return proxyAgent;
+  }
+};
+const proxyAgent = (parsedUrl: URL) => {
+  const protocol = parsedUrl.protocol;
+  if (protocol === 'https') {
+    return new HttpsProxyAgent(process.env.HTTPS_PROXY as string);
+  } else {
+    return new HttpsProxyAgent(process.env.HTTP_PROXY as string);
```

Comment:
sorry for my noob question, but will this work with work on http? There appears to be an `http-proxy-agent` module, should you be using that?

Reply:
lol this is my noob mistake - I wish there was an abstracted version of this as a package. I will just take this for inspiration... https://github.com/touv/node-fetch-with-proxy/blob/master/src/index.js#L53